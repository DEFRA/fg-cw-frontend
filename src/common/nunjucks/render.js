import camelCase from "lodash/camelCase.js";
import { environment } from "./nunjucks.js";

/**
 * @param {string} name
 * @param {object} params
 * @param {string} [block]
 */
export const render = (name, params, block) => {
  const macro = {
    path: `${name}/macro.njk`,
    name: camelCase(name.split("-")),
    params: JSON.stringify(params, null, 2),
  };

  const callComponent = block
    ? `{%- call ${macro.name}(${macro.params}) -%}${block}{%- endcall -%}`
    : `{{- ${macro.name}(${macro.params}) -}}`;

  return environment.renderString(
    `{%- from "${macro.path}" import ${macro.name} -%}
    ${callComponent}
    `,
    {},
  );
};
