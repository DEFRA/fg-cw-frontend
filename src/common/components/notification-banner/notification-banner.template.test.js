import { describe, expect, test } from "vitest";
import { render } from "../../nunjucks/render.js";

describe("notification-banner", () => {
  test("renders with heading and simple text", () => {
    const component = render("notification-banner", {
      heading: "Success",
      text: "Operation completed successfully",
    });

    expect(component).toMatchSnapshot();
  });

  test("renders with heading and complex HTML", () => {
    const component = render("notification-banner", {
      heading: "Success",
      html: '<a class="govuk-notification-banner__link" href="/cases/123">Case ID 123</a> has been assigned to John Doe',
    });

    expect(component).toMatchSnapshot();
  });

  test("renders with text only (no heading)", () => {
    const component = render("notification-banner", {
      text: "Simple notification message",
    });

    expect(component).toMatchSnapshot();
  });

  test("renders with HTML only (no heading)", () => {
    const component = render("notification-banner", {
      html: '<a href="/link">Click here</a> for more information',
    });

    expect(component).toMatchSnapshot();
  });

  test("HTML takes precedence over text when both provided", () => {
    const component = render("notification-banner", {
      heading: "Test",
      text: "This should be ignored",
      html: "<strong>This HTML should be used</strong>",
    });

    expect(component).toMatchSnapshot();
  });

  test("renders with different banner type", () => {
    const component = render("notification-banner", {
      heading: "Warning",
      text: "This is a warning message",
      type: "warning",
    });

    expect(component).toMatchSnapshot();
  });

  test("does not render when no content provided", () => {
    const component = render("notification-banner", {
      heading: "Empty",
    });

    expect(component.trim()).toBe("");
  });

  test("does not render when params is undefined", () => {
    const component = render("notification-banner", undefined);

    expect(component.trim()).toBe("");
  });
});
