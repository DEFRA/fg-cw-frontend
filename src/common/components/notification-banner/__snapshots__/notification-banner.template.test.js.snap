// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`notification-banner > HTML takes precedence over text when both provided 1`] = `
"

  <div class="govuk-notification-banner govuk-notification-banner--success" role="alert" aria-labelledby="govuk-notification-banner-title" data-module="govuk-notification-banner">
  <div class="govuk-notification-banner__header">
    <h2 class="govuk-notification-banner__title" id="govuk-notification-banner-title">
      Success
    </h2>
  </div>
  <div class="govuk-notification-banner__content">
    <h3 class="govuk-notification-banner__heading">Test</h3>
          <p class="govuk-body"><strong>This HTML should be used</strong></p>
  </div>
</div>

"
`;

exports[`notification-banner > renders with HTML only (no heading) 1`] = `
"

  <div class="govuk-notification-banner govuk-notification-banner--success" role="alert" aria-labelledby="govuk-notification-banner-title" data-module="govuk-notification-banner">
  <div class="govuk-notification-banner__header">
    <h2 class="govuk-notification-banner__title" id="govuk-notification-banner-title">
      Success
    </h2>
  </div>
  <div class="govuk-notification-banner__content">
    <p class="govuk-body"><a href="/link">Click here</a> for more information</p>
  </div>
</div>

"
`;

exports[`notification-banner > renders with different banner type 1`] = `
"

  <div class="govuk-notification-banner" role="region" aria-labelledby="govuk-notification-banner-title" data-module="govuk-notification-banner">
  <div class="govuk-notification-banner__header">
    <h2 class="govuk-notification-banner__title" id="govuk-notification-banner-title">
      Important
    </h2>
  </div>
  <div class="govuk-notification-banner__content">
    <h3 class="govuk-notification-banner__heading">Warning</h3>
          <p class="govuk-body">This is a warning message</p>
  </div>
</div>

"
`;

exports[`notification-banner > renders with heading and complex HTML 1`] = `
"

  <div class="govuk-notification-banner govuk-notification-banner--success" role="alert" aria-labelledby="govuk-notification-banner-title" data-module="govuk-notification-banner">
  <div class="govuk-notification-banner__header">
    <h2 class="govuk-notification-banner__title" id="govuk-notification-banner-title">
      Success
    </h2>
  </div>
  <div class="govuk-notification-banner__content">
    <h3 class="govuk-notification-banner__heading">Success</h3>
          <p class="govuk-body"><a class="govuk-notification-banner__link" href="/cases/123">Case ID 123</a> has been assigned to John Doe</p>
  </div>
</div>

"
`;

exports[`notification-banner > renders with heading and simple text 1`] = `
"

  <div class="govuk-notification-banner govuk-notification-banner--success" role="alert" aria-labelledby="govuk-notification-banner-title" data-module="govuk-notification-banner">
  <div class="govuk-notification-banner__header">
    <h2 class="govuk-notification-banner__title" id="govuk-notification-banner-title">
      Success
    </h2>
  </div>
  <div class="govuk-notification-banner__content">
    <h3 class="govuk-notification-banner__heading">Success</h3>
          <p class="govuk-body">Operation completed successfully</p>
  </div>
</div>

"
`;

exports[`notification-banner > renders with text only (no heading) 1`] = `
"

  <div class="govuk-notification-banner govuk-notification-banner--success" role="alert" aria-labelledby="govuk-notification-banner-title" data-module="govuk-notification-banner">
  <div class="govuk-notification-banner__header">
    <h2 class="govuk-notification-banner__title" id="govuk-notification-banner-title">
      Success
    </h2>
  </div>
  <div class="govuk-notification-banner__content">
    <p class="govuk-body">Simple notification message</p>
  </div>
</div>

"
`;
